src/daft-core/src/array/ops/from_arrow.rs

```diff
@@ -30,7 +30,12 @@ where
 {
     fn from_arrow(field: FieldRef, arrow_arr: Box<dyn arrow2::array::Array>) -> DaftResult<Self> {
         let data_array_field = Arc::new(Field::new(field.name.clone(), field.dtype.to_physical()));
-        let physical_arrow_arr = arrow_arr.to_type(data_array_field.dtype.to_arrow()?);
+        let physical_arrow_arr = match field.dtype {
+            // TODO: Consolidate Map to use the same .to_type conversion as other logical types
+            // Currently, .to_type is not implemented for Map in Arrow2, so the conversion is done instead in the from_arrow implementation for ListArray
+            DataType::Map(..) => arrow_arr,
```

Comment:
Is there no `.to_physical()` or equivalent implemented on the arrow2 Map array that could help us do this?

Reply:
The physical type of MapArray is Map in arrow2, but for us it's List.

under the hood, to_type is something that Clark implemented in arrow2:
<img width="886" alt="Screenshot 2024-02-29 at 1 21 55 PM" src="https://github.com/Eventual-Inc/Daft/assets/77712970/d34a614b-4f3e-408b-b614-2bbf07d105d9">
<img width="972" alt="Screenshot 2024-02-29 at 1 22 21 PM" src="https://github.com/Eventual-Inc/Daft/assets/77712970/8e85dbd0-0bdd-4b84-8d5f-96cec14aed9c">

which literally just changes the type attribute if the physical types match, but they don't for MapArray